app/Console/Commands/WaitingLists/ProcessPendingRemovals.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\Console\Commands\WaitingLists;
+
+use App\Console\Commands\Command;
+use App\Events\Training\AccountRemovedFromWaitingListDueToActivity;
+use App\Events\Training\AccountWithinFiveDaysOfWaitingListRemoval;
+use App\Models\Training\WaitingList;
+use Carbon\Carbon;
+
+class ProcessPendingRemovals extends Command
+{
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'waitinglists:processpendingremovals';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Processing waiting list accounts marked for automatic removal';
+
+    /**
+     * Executes all necessary console commands.
+     */
+    public function handle()
+    {
+        WaitingList::all()->filter(function ($waitingList) {
+            return $waitingList->isAtcList();
+        })->each(function (WaitingList $waitingList) {
+            $waitingList->accounts
+            ->filter(function ($account) {
+                return $account->pivot->current_status->name == 'Active';
+            })->filter(function ($account) {
```

Comment:
Another example of where this logic should be abstracted to a model method.

Reply:
Hi @AxonC 

I am struggling with this one a little - can you provide an example of the logic?

I don't work with Laravel in my day-to-day so lack an in-depth knowledge of it at this level. 

The main issue I am facing is that as far as I can see, to add a model method such as `activeAccountsPendingRemoval()` to replace the following two filters:

```
->filter(function ($account) {
    return $account->pivot->current_status->name == 'Active';
})->filter(function ($account) {
    return $account->pivot->pending_removal?->status == 'Pending';
})
```

I would need to add it to the `Account` model, not the `WaitingListAccount` model. Adding it to the `WaitingListAccount` seems like the correct place, but is then not accessible in the query builder as far as I can see due to it being on a pivot and not the parent model.

I believe I could do something like:

```
->filter(function ($account) {
    return $account->pivot->activeAccountsPendingRemoval();
})
```

But using a filter appears to defeat the point of the change.